awx/ui_next/src/components/ListHeader/ListHeader.jsx

```diff
@@ -38,60 +37,59 @@ class ListHeader extends React.Component {
-    this.pushHistoryState(params);
+    const params = parseQueryString(qsConfig, location.search);
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      ...mergeParams(params, { [key]: value }),
+      page: 1,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleReplaceSearch(key, value) {
     const { location, qsConfig } = this.props;
-    const oldParams = parseQueryString(qsConfig, location.search);
-    this.pushHistoryState(replaceParams(oldParams, { [key]: value }));
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      [key]: value,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleRemove(key, value) {
     const { location, qsConfig } = this.props;
-    let oldParams = parseQueryString(qsConfig, location.search);
-    if (parseInt(value, 10)) {
-      oldParams = removeParams(qsConfig, oldParams, {
-        [key]: parseInt(value, 10),
-      });
-    }
-    this.pushHistoryState(removeParams(qsConfig, oldParams, { [key]: value }));
+    const oldParams = parseQueryString(qsConfig, location.search);
+    const updatedParams = removeParams(qsConfig, oldParams, {
+      [key]: value,
+    });
+    const qs = replaceNamespacedParams(
+      qsConfig,
+      location.search,
+      updatedParams
+    );
+    this.pushHistoryState(qs);
   }
 
   handleRemoveAll() {
-    // remove everything in oldParams except for page_size and order_by
     const { location, qsConfig } = this.props;
     const oldParams = parseQueryString(qsConfig, location.search);
-    const oldParamsClone = { ...oldParams };
-    delete oldParamsClone.page_size;
-    delete oldParamsClone.order_by;
-    this.pushHistoryState(removeParams(qsConfig, oldParams, oldParamsClone));
+    Object.keys(oldParams).forEach(key => {
+      oldParams[key] = null;
+    });
+    const qs = replaceNamespacedParams(qsConfig, location.search, oldParams);
+    this.pushHistoryState(qs);
   }
 
   handleSort(key, order) {
     const { location, qsConfig } = this.props;
-    const oldParams = parseQueryString(qsConfig, location.search);
-    this.pushHistoryState(
-      replaceParams(oldParams, {
-        order_by: order === 'ascending' ? key : `-${key}`,
-        page: null,
-      })
-    );
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      order_by: order === 'ascending' ? key : `-${key}`,
+      page: null,
+    });
+    this.pushHistoryState(qs);
   }
 
-  pushHistoryState(params) {
-    const { history, qsConfig } = this.props;
+  pushHistoryState(queryString) {
+    const { history } = this.props;
     const { pathname } = history.location;
-    const nonNamespacedParams = parseQueryString({}, history.location.search);
-    const encodedParams = encodeNonDefaultQueryString(
-      qsConfig,
-      params,
-      nonNamespacedParams
-    );
-    history.push(encodedParams ? `${pathname}?${encodedParams}` : pathname);
+    history.push(queryString ? `${pathname}?${queryString}` : pathname);
```

Comment:
the idea here was to encode only those params that were non-default.  things set as default params (both key and value) would be stripped out of the URL.  And finally, there was trickiness with encoding that things needed to be done a certain way to prevent double encoding (which could take things like spaces and encode them twice, resulting in something invalid).  On the latter point, I would manually test that adding keys with spaces is not broken.

All that being said, I think you moved all of this functionality to replacedNamespacedParams, so I think all of it should be working smoothly.

Reply:
yeah, this all moved into the new util function. I confirmed queries with a space still behave (though both before and after these changes, spaces appear in the URL as a space instead of the encoded `%20`, which might be worth fixing at some point